### PR TITLE
Fix broken Makefile RELEASE_STREAM reporting

### PR DESCRIPTION
**Pull request by bcreane**
## Description
Calico/node's Makefile does not print the correct component versions when overriding the `RELEASE_STREAM` variable. For example, `RELEASE_STREAM=master make calico/node` reports the wrong versions since the versions are evaluated during the first pass.

This PR evaluates the version info in the second pass, and now the reported versions are correct.

Also prettified the version for easy reading and adding a block the `release` target until someone hits <return> (or Cntrl-C).

```

```
## Release Note

```release-note
None required
```

### CODE REVIEW

File: calico_node/Makefile [hunk 1]
```diff
@@ -678,3 +665,18 @@ help: # Some kind of magic from https://gist.github.com/rcmachado/af3db315e31383
...
+	$(info CALICO_GIT_VER=$(CALICO_GIT_VER))
+	$(info BIRD_VER=$(BIRD_VER))
+	$(info CALICOCTL_VER=$(CALICOCTL_VER))
+	$(info CONFD_VER=$(CONFD_VER))
+	$(info CNI_VER=$(CNI_VER))
+	$(info FELIX_VER=$(FELIX_VER))
+	$(info GOBGPD_VER=$(GOBGPD_VER))
+	$(info KUBE_CONTROLLERS_VER=$(KUBE_CONTROLLERS_VER))
+	$(info LIBNETWORK_PLUGIN_VER=$(LIBNETWORK_PLUGIN_VER))
+	$(info TYPHA_VER=$(TYPHA_VER))
```
Comment (neiljerram) [inline calico_node/Makefile]: Not sure this is right.  I _think_ `$(info)` is magic that is executed as the Makefile is parsed, not something that can expand into target actions.

But on the other hand you've presumably tried this out - what are your observations?

**Comment by bcreane** [comment]
when I make the target, it helpfully prints out the versions its pulling from _data/versions.yml:
```
make check-versions 
RELEASE_STREAM        = v3.0
CALICO_VER            = v3.0.0-alpha1-rc1
CALICO_GIT_VER        = v3.0.0-alpha1-rc1-3-gf68db25-dirty
BIRD_VER              = v0.3.1
CALICOCTL_VER         = v2.0.0-alpha1-rc2
CONFD_VER             = null
CNI_VER               = v2.0.0-alpha1-rc1
FELIX_VER             = 3.0.0-alpha1-rc1
GOBGPD_VER            = v0.2.1
KUBE_CNTRLRS_VER      = v2.0.0-alpha1-rc1
LIBNETWORK_PLUGIN_VER = v1.1.0
TYPHA_VER             = v0.6.0-alpha1-rc1
upstream              = f68db25087b054f332cfd1febf4dd1dd42ecef04
local                 = f68db25087b054f332cfd1febf4dd1dd42ecef04
make: Nothing to be done for 'check-versions'.
```

**Comment by neiljerram** [comment]
@bcreane I think the `Nothing to be done for check-versions` output is a clue, and supports my hypothesis that the printing really isn't associated with that particular target.

What happens if you do `make <any other target>`? I'm guessing that you will still see the versions. If that's right, can we move them back to where there were before, but still make the other improvements that you've made here?

**Comment by bcreane** [comment]
Hey @neiljerram - making other targets doesn't show the versions, so aside from the `Nothing to be done ...`, it seems to work as expected. I thought the `Nothing ...` is an effect of not updating any actual targets; it still executes the `info` statements, but can't find a target to update.

An alternative to having an explicit target that just prints stuff is to move these `info`s into the `release` target. It would be nice to block building the release until we have time to inspect the settings and manually indicate "yes" to proceeding with the release.

Have you seen a good approach for incorporating a `read` (y/n) verification in Makefile?

**Comment by neiljerram** [comment]
OK, cool, so it does work as a per-target thing then.

Another thought, though... at the moment I think those versions are displayed for any make invocation.  Isn't that actually quite useful?  (For example, for me at the moment running `make st`)  I'm not sure how it would help to hide them inside just some specific targets.

Regarding a y/n query - look for "bash -c read" in the Felix Makefile, for one possibility.

**Comment by neiljerram** [comment]
Hi @bcreane - just reviewed this again, and I'm sorry that my thoughts have been arriving as a kind of drip-feed...

Stepping back, I don't see at all what the benefit is of the proposed versions target.  I like what we have now, namely that the versions are printed out *every* time you run make.  Why do you think we should change this?

**Comment by bcreane** [comment]
Hey Neil, my main goal is to print the versions while building a release and then pause until the user hits <return> (or breaks). Otherwise the versions scroll past and there's no opportunity to inspect it.

I chose to make this a make target so that only targets that are affected by RELEASE_STREAM variables can print out this info - it doesn't make sense to print it for `clean` for example.

**Comment by bcreane** [comment]
going to look some more at this and reopen later

**Pull request closed without merging**